src/consensus/sync_queue.cpp

```diff
@@ -0,0 +1,44 @@
+#include "sync_queue.hpp"
+
+#include "consensus/pbft_chain.hpp"
+#include "consensus/sync_block.hpp"
+#include "dag/dag_block.hpp"
+#include "transaction_manager/transaction.hpp"
+
+namespace taraxa {
+
+uint64_t SyncBlockQueue::getPeriod() const {
+  std::shared_lock lock(sync_queue_access_);
+  return sync_period_;
+}
+
+size_t SyncBlockQueue::size() const {
+  std::shared_lock lock(sync_queue_access_);
+  return sync_queue_.size();
+}
+
+void SyncBlockQueue::clear() {
+  std::unique_lock lock(sync_queue_access_);
+  sync_period_ = 0;
+  sync_queue_.clear();
+}
+
+bool SyncBlockQueue::push(SyncBlock &&block, dev::p2p::NodeID const &node_id, uint64_t max_pbft_size) {
+  const auto period = block.pbft_blk->getPeriod();
+  std::unique_lock lock(sync_queue_access_);
+  if (period != std::max(sync_period_, max_pbft_size) + 1) {
```

Comment:
not sure if it ican actually happen but according to this code, it might be possible that  `sync_period_ != 0 (this means sync_queue_ is not empty)` and `max_pbft_size > sync_period_.` In such case it is pointless to push into queue on top of the data that is there because such data are not valid. In case this is true, we could clear sync_queue

Reply:
because of previous discussion this needs to be like this

Reply:
no this is different. If `max_pbft_size > sync_period_` it means sync_queue has outdated data which will be rejected anyway

Reply:
nope, let's say you sync up to period 5, then you will receive "normally" blocks up to period 10 and then sync again. `sync_period_` will not be cleared but `max_pbft_size` will be higher. That's what happened on testnet

Reply:
ok update, this is what I meant: 

```
if (max_pbft_size > sync_period_ && !sync_queue.empty()) {
 sync_queue.clear();
}
```